core/src/main/java/net/sourceforge/jnlp/Launcher.java

```diff
@@ -137,33 +110,12 @@ private Launcher(LaunchHandler handler) {
      *
      * @param handler the handler to use or null for no handler.
      * @param policy the update policy to use or null for default policy.
+     * @param exitOnFailure exit if there is an error (usually default, but false when being used from the plugin)
      */
-    private Launcher(LaunchHandler handler, UpdatePolicy policy) {
-        if (policy == null)
-            policy = JNLPRuntime.getDefaultUpdatePolicy();
-
-        this.handler = handler;
-        this.updatePolicy = policy;
-
-    }
-
-    /**
-     * Sets the update policy used by launched applications.
-     * @param policy to be used for resources
-     */
-    public void setUpdatePolicy(UpdatePolicy policy) {
-        if (policy == null) {
-            throw new IllegalArgumentException("Update policy cannot be null.");
-        }
-
-        this.updatePolicy = policy;
-    }
-
-    /**
-     * @return the update policy used when launching applications.
-     */
-    public UpdatePolicy getUpdatePolicy() {
-        return updatePolicy;
+    public Launcher(final LaunchHandler handler, final UpdatePolicy policy, boolean exitOnFailure) {
+        this.handler = (handler == null) ? JNLPRuntime.getDefaultLaunchHandler() : null;
```

Comment:
I believe this should not assign `null` but instead `handler`
Also parentheses can be removed, but maybe leaving them in is better for readability, I'm undecided